feat(vs): Add curl example

goock opened the pull request:
This PR introduces curl examples that operate on VS, VM, and VMI RESTful APIs.

## TODO:

- [ ] - add documentation based on readme

goock commented:
> I didn't spend a bunch of time figuring out why, but I wasn't able to get this to work. It loops without creating a VS as far as I can tell (Ubuntu 20.04 w/ `jq` and `curl` installed.
> 
> To be honest, I think a GitBook page with some simple `curl` examples, an explainer on how to get a token out of a `kubeconfig`, and a link to the [kubevirt API reference](https://kubevirt.io/api-reference/v0.13.7/operations.html) is all we really need here.
> 
> ```
> TOKEN=<redacted> NAMESPACE=tenant-coreweave-cpierre ./run.sh 
> 
> ------------- CREATE VS -----------------------------------
> 
> ------------- START VS -----------------------------------
> 
> ------------- WAIT FOR VMI UNTIL RUNNING ------------------
> Status:, Started:
> ```

Believe me, I thought exactly the same that a simple reference to API and how to get the token would be enough but this example is based on a conversation with users and their problems.

@ChandonPierre I will appreciate is you can run the script with `set -x` enabled and paste the output. It's really odd it does not work for you.

ChandonPierre commented:
> @ChandonPierre I will appreciate is you can run the script with `set -x` enabled and paste the output. It's really odd it does not work for you.
> BTW: the example was prepared for a cloud app so you need to use the cloud app's token and namespace. You may need to change `APISERVER=k8s.ord1.staging.coreweave.com` if you do not have a prod account.

Okay, looks like the script was set to staging but I was using a prod token. Changed to prod API server and it worked 👍

goock commented:
> > @ChandonPierre I will appreciate is you can run the script with `set -x` enabled and paste the output. It's really odd it does not work for you.
> > BTW: the example was prepared for a cloud app so you need to use the cloud app's token and namespace. You may need to change `APISERVER=k8s.ord1.staging.coreweave.com` if you do not have a prod account.
> 
> Okay, looks like the script was set to staging but I was using a prod token. Changed to prod API server and it worked +1

Indeed, It was set to staging, I changed that to prod, thanks to you Chandon.
Thank you for verifying the script.

microcosem commented on virtual-server/examples/curl/run.sh:
```diff
@@ -0,0 +1,220 @@
+APISERVER=k8s.ord1.coreweave.com
+
+if [ -z ${NAMESPACE} ] || [ -z ${TOKEN} ]; then
+  echo "NAMESPACE and TOKEN variable must be set: TOKEN=<token> NAMESPACE=<namespace> ./run.sh"
```
@YitzyD I addressed a lot of these in my commit but those last two I'll incorporate. :)

microcosem reviewed (approved):
Approving my own changes.

goock commented on virtual-server/examples/curl/run.sh:
```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+# Enable to trace the script
+# set -x
```
I cannot set them:
- `-e` terminates on any none 0 exits, including conditions `[...]`
- `-u` complains about unused `NAMESPACE` or `TOKEN` variables and the message won't have a chance to be displayed.

microcosem commented:
> @microcosem @YitzyD Thank you for correcting me. To make my task easier I based on the node.js https://github.com/coreweave/kubernetes-cloud/tree/master/virtual-server/examples/nodejs, we might need to review the rest of them as well, to be coherent across all examples.

I was wondering about this, thanks for mentioning it. I'll add a task to my list to go over the others that already exist! 🙂

goock commented:
> > @microcosem @YitzyD Thank you for correcting me. To make my task easier I based on the node.js [https://github.com/coreweave/kubernetes-cloud/tree/master/virtual-server/examples/nodejs](https://github.com/coreweave/kubernetes-cloud/tree/master/virtual-server/examples/nodejs?rgh-link-date=2022-06-14T16%3A39%3A15Z), we might need to review the rest of them as well, to be coherent across all examples.
> 
> I was wondering about this, thanks for mentioning it. I'll add a task to my list to go over the others that already exist! slightly_smiling_face

Actually, this one https://github.com/coreweave/kubernetes-cloud/pull/57 is queuing for review. I added a new feature there, so there is a chance to fix the description in the README and GitBook.

microcosem commented:
> > > @microcosem @YitzyD Thank you for correcting me. To make my task easier I based on the node.js [https://github.com/coreweave/kubernetes-cloud/tree/master/virtual-server/examples/nodejs](https://github.com/coreweave/kubernetes-cloud/tree/master/virtual-server/examples/nodejs?rgh-link-date=2022-06-14T16%3A39%3A15Z), we might need to review the rest of them as well, to be coherent across all examples.
> > 
> > 
> > I was wondering about this, thanks for mentioning it. I'll add a task to my list to go over the others that already exist! slightly_smiling_face
> 
> Actually, this one #57 is queuing for review. I added a new feature there, so there is a chance to fix the description in the README and GitBook.

I've added it to the Project and will be reviewing and updating!

goock commented:
@microcosem Since the script works and the readme is fine, I merged the branch. Please let me know how would you like to present the curl examples into the GitBook? if you have any ideas. I could be too pragmatic and put it along with other languages.
Curl is the most straightforward way of presenting RESTful API access. It can be used in many other languages.
